Randomizer - Shuffle Roll

TheLynk opened the pull request:
Add new setting randomizer "Shuffle Roll", shuffle ability do a roll for Link
Logic update for Tree and Break Crates

I am not against feedback or ideas for the custom model
And for All language Translation 
And for logic

Thanks @PABessero for French Translation

<!--- section:artifacts:start -->
### Build Artifacts
  - [soh-windows.zip](https://nightly.link/TheLynk/Shipwright/actions/artifacts/5271655265.zip)
  - [soh-linux.zip](https://nightly.link/TheLynk/Shipwright/actions/artifacts/5271688318.zip)
  - [soh-mac.zip](https://nightly.link/TheLynk/Shipwright/actions/artifacts/5272324849.zip)
<!--- section:artifacts:end -->

TheLynk committed:
Add Shuffle Roll (V1)

TODO New model for items

TheLynk committed:
Clang Format

TheLynk committed:
New Model For Roll

TheLynk committed:
Rename Custom Model in Code

TheLynk committed:
Add more translation

TheLynk committed:
Tweak Translation

TheLynk committed:
Update soh/soh/Enhancements/game-interactor/vanilla-behavior/GIVanillaBehavior.h

Co-authored-by: Philip Dubé <serprex@users.noreply.github.com>

TheLynk committed:
Tweak code | Thx @serprex

PABessero commented on soh/soh/Enhancements/randomizer/3drando/hint_list/hint_list_item.cpp:
```diff
@@ -2100,6 +2100,8 @@ void StaticData::HintTable_Init_Item() {
                                                { CustomMessage("a master unlocker", /*german*/ "ein Meisterentsperrer", /*french*/ "un Kit de Déverrouillage") });
                                                 // /*spanish*/un desbloqueador maestro
 
+    hintTextTable[RHT_ROLL] = HintText(CustomMessage("the ability to roll", /*german*/"la capacité de faire roulade", /*french*/"die Fähigkeit zu rollen"));
```
French and german are inverted

PABessero replied:
"de faire roulade" --> "de rouler" / "de faire des roulades" / "de roulader" (oui, ce dernier est francais d'apres le dictionnaire)

PABessero commented on soh/soh/Enhancements/randomizer/randomizer.cpp:
```diff
@@ -5812,6 +5812,9 @@ void Randomizer::CreateCustomMessages() {
         GIMESSAGE(RG_BRONZE_SCALE, ITEM_SCALE_SILVER, "You got the %rBronze Scale%w!&The power of buoyancy is yours!",
                   "Du hast die %rBronzene Schuppe%w&erhalten! Die Fähigkeit zu&Schwimmen ist nun dein!",
                   "Vous obtenez l'%rÉcaille de Bronze%w!&Le pouvoir de la flottabilité est&à vous!"),
```
Flottabilité --> flottaison

TheLynk replied:
fait ta propre PR pour tweak ça

PABessero commented on soh/soh/Enhancements/randomizer/randomizer.cpp:
```diff
@@ -5812,6 +5812,9 @@ void Randomizer::CreateCustomMessages() {
                   "Vous obtenez l'%rÉcaille de Bronze%w!&Le pouvoir de la flottabilité est&à vous!"),
+        GIMESSAGE(RG_ROLL, ITEM_SCALE_SILVER, "You got the %rAbility do a roll%w!",
+                  "Du hast die Fähigkeit, %reinen Wurf&zu machen%w!",
+                  "Vous obtenez la %rCapacité à faire&des roulade%w!"),
```
"de faire des roulades"

PABessero reviewed (changes requested):
Review for orthographic typos

TheLynk committed:
Update soh/soh/Enhancements/randomizer/logic.cpp

Co-authored-by: Philip Dubé <serprex@users.noreply.github.com>

TheLynk committed:
Fix translation

TheLynk committed:
Re Add All the branch (but no Shuffle Silver Rupee) | V1.0

TheLynk committed:
Add Branch "Shuffle Roll" | Version : 2.0

TheLynk committed:
Fix Text in Item tracker and Logic for ZR | V:2.1

TheLynk committed:
Fix Logic in KF

TheLynk committed:
Fix Logic in KF | 2.2

TheLynk committed:
Add Branch add the ability to Randomize Music (Seeded) | V:3.0

TheLynk committed:
Add branch "Invisible Navi option" | V:4.0

TheLynk committed:
Add branch "Rando: Add Triforce Hunt GBK setting" | V:5.0

TheLynk committed:
Add branch "Add Edit buttons for custom presets" | V6.0

TheLynk committed:
Add branch "Check tracker improvements" | V:7.0

TheLynk committed:
Add branch "Improvements to Custom Kaleido Menu" | V:8.0

TheLynk committed:
Fix Logic

TheLynk committed:
Fix Logic | V:8.1

TheLynk committed:
Fix Logic Shuffle Bean Souls | V:8.2

TheLynk committed:
Merge branch 'develop' into Shuffle-Roll

TheLynk committed:
Update model for Shuffle Roll | V:8.3

TheLynk committed:
Fix Merge And New Model for Roll

TheLynk committed:
Merge branch 'develop' into V3-Anchor-Fusion-All-New-Option

TheLynk committed:
Fix Update Auto Merge | V:9.0

TheLynk committed:
Fix crash Logic | V:9.1

TheLynk committed:
Add branch "rocksanity" | V:10.0

TheLynk committed:
clang-format

TheLynk committed:
Fix Logic | V:10.1

PABessero committed:
Merge branch 'develop' into V3-Anchor-Fusion-All-New-Option

good luck

TheLynk committed:
Major Update | V:11

TheLynk committed:
Merge branch 'develop' into V3-Anchor-Fusion-All-New-Option

TheLynk committed:
Good Luck Update | V:12

small issue is found is if open chest is not shuffle the randomizer run is imbattable

TheLynk committed:
Merge remote-tracking branch 'upstream/develop' into Shuffle-Roll

Part 1

TheLynk committed:
Merge remote-tracking branch 'upstream/develop' into Shuffle-Roll

Part 2

TheLynk committed:
Add Animation Capture Tool.cpp & remove useless code in draw.cpp

TheLynk committed:
Update animation_capture_tool.cpp

TheLynk committed:
Update soh/soh/Enhancements/randomizer/animation_capture_tool.cpp

Co-authored-by: Philip Dubé <serprex@users.noreply.github.com>

TheLynk committed:
Merge branch 'develop' into Shuffle-Roll

TheLynk committed:
Clang Format

TheLynk committed:
Fix crash and Add Ice traps Text for Roll

TheLynk committed:
Add Roll in Kaleido Tracker

TheLynk committed:
Merge branch 'HarbourMasters:develop' into develop (Manuel)

TheLynk committed:
Add Roll in Item Tracker

TheLynk committed:
To many addition | V:13

TheLynk committed:
To many addition | V:13

TheLynk committed:
Update Logic | V:13.1

TheLynk committed:
Merge branch 'V3-Anchor-Fusion-All-New-Option' into Shuffle-Roll

TheLynk committed:
Revert "Merge branch 'V3-Anchor-Fusion-All-New-Option' into Shuffle-Roll"

This reverts commit a8fee8ed4acc89444c6a102908c0dd06b198e56a, reversing
changes made to 1f2c3832762ae2a50683c781e1a03958ac648dc7.

TheLynk committed:
Add Missing Line
